ca.go

```diff
@@ -18,19 +20,29 @@ type CA struct {
 	// may be set to something like `nil`.
 	Rand io.Reader
 
-	// Wrapped `crypto.Signer` to preform OpenSSH CA operations.
-	Signer ssh.Signer
+	// Function to choose which signer to use based on request information
+	ChooseSigner func(events.APIGatewayProxyRequest) (crypto.Signer, error)
 }
 
 // Sign an SSH Certificate template (with `Key` set), and return the
 // certificate.
-func (s CA) Sign(template ssh.Certificate) (*ssh.Certificate, error) {
+func (s CA) Sign(template ssh.Certificate, event events.APIGatewayProxyRequest) (*ssh.Certificate, error) {
+	signer, err := s.ChooseSigner(event)
```

Comment:
It may be worth having the default be sane here, `if s.ChooseSigner == nil` use `DefaultSigner` or something. That or check if nil and return an error.